### PR TITLE
added telemetry api calls

### PR DESCRIPTION
**Pull request by RoyUP9**

### CODE REVIEW

File: agent/pkg/providers/stats_provider.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+type GeneralStats struct {
+	EntriesCount        int
+	FirstEntryTimestamp int
+	LastEntryTimestamp  int
+}
+
+var (
+	generalStats GeneralStats
```
Comment (IgorGov) [inline agent/pkg/providers/stats_provider.go:15]: generalStats := GeneralStats{}

File: agent/pkg/providers/stats_provider.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+func init() {
```
Comment (IgorGov) [inline agent/pkg/providers/stats_provider.go:18]: redundant

File: cli/cmd/tapRunner.go [hunk 3]
```diff
@@ -249,8 +250,12 @@ func updateMizuTappers(ctx context.Context, kubernetesProvider *kubernetes.Provi
 	return nil
 }
 
-func cleanUpMizuResources(kubernetesProvider *kubernetes.Provider) {
+func cleanUpMizu(kubernetesProvider *kubernetes.Provider) {
+	telemetry.ReportAPICalls(config.Config.Tap.GuiPort)
+	cleanUpMizuResources(kubernetesProvider)
```
Comment (IgorGov) [inline cli/cmd/tapRunner.go:255]: lets export a function of dump logs if needed and call it here

File: cli/telemetry/telemetry.go [hunk 4]
```diff
@@ -5,39 +5,107 @@ import (
...
 	"net/http"
 )
 
 const telemetryUrl = "https://us-east4-up9-prod.cloudfunctions.net/mizu-telemetry"
 
 func ReportRun(cmd string, args interface{}) {
-	if !config.Config.Telemetry {
-		logger.Log.Debugf("not reporting due to config value")
+	if !shouldRunTelemetry() {
+		logger.Log.Debugf("not reporting telemetry")
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:19]: return

File: cli/telemetry/telemetry.go [hunk 5]
```diff
@@ -5,39 +5,107 @@ import (
...
+		logger.Log.Debugf("%v", err)
 		return
 	}
 
+	logger.Log.Debugf("successfully reported telemetry for cmd %v", cmd)
+}
+
+func ReportAPICalls(mizuPort uint16) {
+	if !shouldRunTelemetry() {
+		logger.Log.Debugf("not reporting telemetry")
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:43]: return

File: cli/telemetry/telemetry.go [hunk 6]
```diff
@@ -5,39 +5,107 @@ import (
...
+		"component":      "mizu_cli",
+		"buildTimestamp": mizu.BuildTimestamp,
+		"branch":         mizu.Branch,
+		"version":        mizu.SemVer,
+		"cmd":            cmd,
+		"args":           string(argsBytes),
+	}
+
+	if err := sendTelemetry(argsMap); err != nil {
+		logger.Log.Debugf("%v", err)
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:34]: logger.Log.Debug(err)

File: cli/telemetry/telemetry.go [hunk 7]
```diff
@@ -5,39 +5,107 @@ import (
...
+		"buildTimestamp":        mizu.BuildTimestamp,
+		"branch":                mizu.Branch,
+		"version":               mizu.SemVer,
+		"apiCallsCount":         generalStats["EntriesCount"],
+		"firstAPICallTimestamp": generalStats["FirstEntryTimestamp"],
+		"lastAPICallTimestamp":  generalStats["LastEntryTimestamp"],
+	}
+
+	if err := sendTelemetry(argsMap); err != nil {
+		logger.Log.Debugf("%v", err)
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:84]: logger.Log.Debug(err)

File: cli/telemetry/telemetry.go [hunk 8]
```diff
@@ -5,39 +5,107 @@ import (
...
+	if !shouldRunTelemetry() {
+		logger.Log.Debugf("not reporting telemetry")
+	}
+
+	mizuProxiedUrl := kubernetes.GetMizuApiServerProxiedHostAndPath(mizuPort)
+	generalStatsUrl := fmt.Sprintf("http://%s/api/generalStats", mizuProxiedUrl)
+
+	response, requestErr := http.Get(generalStatsUrl)
+	if requestErr != nil {
+		logger.Log.Debugf("failed to get general stats for telemetry, err: %v", requestErr)
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:51]: "ERROR

File: cli/telemetry/telemetry.go [hunk 9]
```diff
@@ -5,39 +5,107 @@ import (
...
+	}
+
+	var generalStats map[string]interface{}
+	if parseErr := json.Unmarshal(data, &generalStats); parseErr != nil {
+		logger.Log.Debugf("failed to parse general stats for telemetry, err: %v", parseErr)
+		return
+	}
+
+	argsMap := map[string]interface{}{
+		"telemetryType":         "APICalls",
```
Comment (IgorGov) [inline cli/telemetry/telemetry.go:73]: lets move these 5 params to sendTelemtry function

File: cli/cmd/tapRunner.go (hunk 3 above)
Comment (RoyUP9) [reply cli/cmd/tapRunner.go:255]: there is not simple solution for it

**Pull request merged**